Documentation/policy/language.rst

```diff
@@ -34,10 +34,16 @@ Labels Based
 ------------
 
 Label-based L3 policy is used to establish policy between endpoints inside the
-cluster managed by Cilium. An endpoint is allowed to talk to another endpoint
-if at least one rule exists which selects the destination endpoint with the
-`EndpointSelector` in the ``endpointSelector`` field and selects the source
-endpoint with the `EndpointSelector` in the ``fromEndpoints`` field.
+cluster managed by Cilium.
+
+Ingress
+~~~~~~~
+
+An endpoint is allowed to receive traffic from another endpoint if at least one
```

Comment:
I wonder if the "send" / "receive" traffic descriptions here are a little too strict if read literally. It's a bit more abstract, the previous wording said "talk to" to imply that if it talks to the other endpoint, the other endpoint can talk back on the same channel, whereas "send traffic" implies unidirectional. But maybe I'm just reading too far into the wording.

Reply:
One counter I have to what you are saying is that we can still have label-based policy without connection tracking enabled; that is, we can disable connection tracking so that in order for an endpoint A to talk to another endpoint B, not only does A need to have an egress policy allowing it to talk to B, B needs a policy allowing ingress from A, and vice versa. I think not specifying whether communication is allowed in the reverse direction is good, because that is related to another configuration option (conntrack).

Reply:
Fair counter, though let me counter counter ( ;D ) with this: What's the default? Shouldn't we document the behaviour as per the default operation of the daemon?

I'm not sure if the conntrack option you're referring to is even documented, and not only that, I honestly see it as a serious power user tweak because it breaks policies in a pretty fundamental way. You have to pretty much rewrite your entire policies to use it, and even then seems painful to me.

Reply:
I disagree. Let's not make the documentation more complicated than it has to be. If we want to document the default behavior of the demon with connection tracking, let's get a separate section out to do so. This section specifically pertains to policy alone, not to daemon operations.